crates/rustc_codegen_spirv/src/linker/mod.rs

```diff
@@ -231,6 +233,11 @@ pub fn link(sess: &Session, mut inputs: Vec<Module>, opts: &Options) -> Result<L
         }
     }
 
+    if opts.destructure {
+        let _timer = sess.timer("link_destructure");
+        destructure_composites::destructure_composites(&mut output);
+    }
```

Comment:
Really silly nit, it might be nice to move this to run right after `mem2reg::mem2reg` (within that for loop) and take a `&mut Function` instead, since it doesn't actually need the entire module to work on (it processes functions individually), would help cache locality and performance to not scan over the module twice.

Reply:
Yes, but I'd like to preserve each feature flag being its own clause in the function - I think double-scanning the module is not the hugest cost, in my eyes, the cleanliness wins.